lhs.js

```diff
@@ -42,6 +47,28 @@ class Lhs {
         }
     }
 
+    when(pred) {
+        this._predicates.push(pred);
```

Comment:
I'm sorry; I should have noticed this earlier. LHSs are immutable so repetitive ones can be factored out. Thus, this should act like TypeLhs.type() and TypeLhs.max() in returning a fresh LHS.

Also, we should probably take another cue from TypeLhs.type() in that later when()s should override earlier ones. This lets us factor out repetition. (Here's [the documentation about similar behavior in the RHS](http://mozilla.github.io/fathom/ruleset.html?highlight=repetition#right-hand-sides).) So the _predicates array can be replaced with a simple value. If we someday want to be able to combine when()s, we could do it through, perhaps, and `andWhen()` method. Or things might change a whole lot before then; it's best not to worry about it unless we need it immediately.